packages/plugin/webpack/src/WebpackPlugin.ts

```diff
@@ -283,25 +284,54 @@ the generated files). Instead, it is ${JSON.stringify(pj.main)}`);
   };
 
   launchRendererDevServers = async (logger: Logger): Promise<void> => {
-    const config = await this.configGenerator.getRendererConfig(this.config.renderer.entryPoints);
-    if (config.length === 0) {
+    const configs = await this.configGenerator.getRendererConfig(this.config.renderer.entryPoints);
+    if (configs.length === 0) {
       return;
     }
 
-    for (const entryConfig of config) {
+    const preloadPlugins: string[] = [];
+    let preloadEntriesWithConfig = 0;
+    for (const entryConfig of configs) {
       if (!entryConfig.plugins) entryConfig.plugins = [];
       entryConfig.plugins.push(new ElectronForgeLoggingPlugin(logger.createTab(`Renderer Target Bundle (${entryConfig.target})`)));
 
+      const filename = entryConfig.output?.filename as string;
+      if (filename.endsWith('preload.js')) {
+        let name = `entry-point-preload-${entryConfig.target}`;
+        if (preloadPlugins.includes(name)) {
+          name = `${name}-${++preloadEntriesWithConfig}`;
+        }
+        entryConfig.plugins.push(new EntryPointPreloadPlugin({ name }));
+        preloadPlugins.push(name);
+      }
+
       entryConfig.infrastructureLogging = {
         level: 'none',
       };
       entryConfig.stats = 'none';
     }
 
-    const compiler = webpack(config);
+    const compiler = webpack(configs);
+
+    const promises = [];
+    for (const preloadPlugin of preloadPlugins) {
+      promises.push(
+        new Promise((resolve, reject) => {
+          compiler.hooks.done.tap(preloadPlugin, (stats) => {
+            if (stats.hasErrors()) {
+              // TODO(georgexu99): add a way to identify preload entries with the same target name ex) preload config entries
+              return reject(new Error(`Compilation errors in the preload: ${stats.toString()}`));
+            }
+            return resolve(undefined);
+          });
+        })
+      );
+    }
```

Comment:
nit/suggestion: you can use `map` here (or you can leave it as is, it works either way 🙂)

```suggestion
    const promises = preloadPlugins.map((preloadPlugin) => {
      return new Promise((resolve, reject) => {
        compiler.hooks.done.tap(preloadPlugin, (stats) => {
          if (stats.hasErrors()) {
            // TODO(georgexu99): add a way to identify preload entries with the same target name ex) preload config entries
            return reject(new Error(`Compilation errors in the preload: ${stats.toString()}`));
          }
          return resolve(undefined);
        });
      });
    });
```